NotGiven Typechecking Compile Crash

## Compiler version

3.2.1 and above

Also fails on 3.2.1 but with a different error.

Logs provided from 3.3.2.

## Minimized code

```Scala
import scala.util.NotGiven

type  GivenOrNot[T] = T | NotGiven[T]

inline def perform[T](using GivenOrNot[T]): T = perform2

inline def perform2[T](using GivenOrNot[T]): T = scala.compiletime.error("err")
```

## Output (click arrow to expand)
<details>

```scala
[error] java.lang.AssertionError: assertion failed
[error] scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:11)
[error] dotty.tools.dotc.core.TypeOps$.dominators$1(TypeOps.scala:248)
[error] dotty.tools.dotc.core.TypeOps$.approximateOr$1(TypeOps.scala:382)
[error] dotty.tools.dotc.core.TypeOps$.orDominator(TypeOps.scala:395)
[error] dotty.tools.dotc.core.Types$OrType.join(Types.scala:3435)
[error] dotty.tools.dotc.core.Types$Type.classSymbol(Types.scala:531)
[error] dotty.tools.dotc.typer.Implicits$ImplicitSearch.<init>(Implicits.scala:1212)
[error] dotty.tools.dotc.typer.Implicits.inferImplicit(Implicits.scala:1060)
[error] dotty.tools.dotc.typer.Implicits.inferImplicit$(Implicits.scala:818)
[error] dotty.tools.dotc.typer.Typer.inferImplicit(Typer.scala:116)
[error] dotty.tools.dotc.typer.Implicits.inferImplicitArg(Implicits.scala:884)
[error] dotty.tools.dotc.typer.Implicits.inferImplicitArg$(Implicits.scala:818)
[error] dotty.tools.dotc.typer.Typer.inferImplicitArg(Typer.scala:116)
[error] dotty.tools.dotc.typer.Typer.implicitArgs$1(Typer.scala:3737)
[error] dotty.tools.dotc.typer.Typer.addImplicitArgs$1(Typer.scala:3773)
[error] dotty.tools.dotc.typer.Typer.adaptNoArgsImplicitMethod$1(Typer.scala:3849)
[error] dotty.tools.dotc.typer.Typer.adaptNoArgs$1(Typer.scala:4038)
[error] dotty.tools.dotc.typer.Typer.adapt1(Typer.scala:4268)
[error] dotty.tools.dotc.typer.Typer.adapt(Typer.scala:3587)
[error] dotty.tools.dotc.typer.Typer.readapt$1(Typer.scala:3598)
[error] dotty.tools.dotc.typer.Typer.adapt1(Typer.scala:4255)
[error] dotty.tools.dotc.typer.Typer.adapt(Typer.scala:3587)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3184)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3188)
[error] dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3300)
[error] dotty.tools.dotc.typer.Typer.$anonfun$57(Typer.scala:2486)
[error] dotty.tools.dotc.inlines.PrepareInlineable$.dropInlineIfError(PrepareInlineable.scala:243)
[error] dotty.tools.dotc.typer.Typer.typedDefDef(Typer.scala:2486)
[error] dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3024)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3111)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3184)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3188)
[error] dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3210)
[error] dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3256)
[error] dotty.tools.dotc.typer.Typer.typedClassDef(Typer.scala:2669)
[error] dotty.tools.dotc.typer.Typer.typedTypeOrClassDef$1(Typer.scala:3036)
[error] dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3040)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3111)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3184)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3188)
[error] dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3210)
[error] dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3256)
[error] dotty.tools.dotc.typer.Typer.typedPackageDef(Typer.scala:2812)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3081)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3112)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3184)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3188)
[error] dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3300)
[error] dotty.tools.dotc.typer.TyperPhase.typeCheck$$anonfun$1(TyperPhase.scala:44)
[error] dotty.tools.dotc.typer.TyperPhase.typeCheck$$anonfun$adapted$1(TyperPhase.scala:54)
[error] scala.Function0.apply$mcV$sp(Function0.scala:42)
[error] dotty.tools.dotc.core.Phases$Phase.monitor(Phases.scala:440)
[error] dotty.tools.dotc.typer.TyperPhase.typeCheck(TyperPhase.scala:54)
[error] dotty.tools.dotc.typer.TyperPhase.runOn$$anonfun$3(TyperPhase.scala:88)
[error] scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error] scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error] scala.collection.immutable.List.foreach(List.scala:333)
[error] dotty.tools.dotc.typer.TyperPhase.runOn(TyperPhase.scala:88)
[error] dotty.tools.dotc.Run.runPhases$1$$anonfun$1(Run.scala:246)
[error] scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error] scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error] scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1321)
[error] dotty.tools.dotc.Run.runPhases$1(Run.scala:262)
[error] dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:270)
[error] dotty.tools.dotc.Run.compileUnits$$anonfun$adapted$1(Run.scala:279)
[error] dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:67)
[error] dotty.tools.dotc.Run.compileUnits(Run.scala:279)
[error] dotty.tools.dotc.Run.compileSources(Run.scala:194)
[error] dotty.tools.dotc.Run.compile(Run.scala:179)
[error] dotty.tools.dotc.Driver.doCompile(Driver.scala:37)
[error] dotty.tools.xsbt.CompilerBridgeDriver.run(CompilerBridgeDriver.java:88)
[error] dotty.tools.xsbt.CompilerBridge.run(CompilerBridge.java:22)
[error] sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:91)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$7(MixedAnalyzingCompiler.scala:193)
[error] scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error] sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:248)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4(MixedAnalyzingCompiler.scala:183)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4$adapted(MixedAnalyzingCompiler.scala:163)
[error] sbt.internal.inc.JarUtils$.withPreviousJar(JarUtils.scala:239)
[error] sbt.internal.inc.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:163)
[error] sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:211)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:534)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:534)
[error] sbt.internal.inc.Incremental$.$anonfun$apply$5(Incremental.scala:180)
[error] sbt.internal.inc.Incremental$.$anonfun$apply$5$adapted(Incremental.scala:178)
[error] sbt.internal.inc.Incremental$$anon$2.run(Incremental.scala:464)
[error] sbt.internal.inc.IncrementalCommon$CycleState.next(IncrementalCommon.scala:116)
[error] sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:56)
[error] sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:52)
[error] sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:263)
[error] sbt.internal.inc.Incremental$.$anonfun$incrementalCompile$8(Incremental.scala:419)
[error] sbt.internal.inc.Incremental$.withClassfileManager(Incremental.scala:506)
[error] sbt.internal.inc.Incremental$.incrementalCompile(Incremental.scala:406)
[error] sbt.internal.inc.Incremental$.apply(Incremental.scala:172)
[error] sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:534)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:488)
[error] sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCompilerImpl.scala:332)
[error] sbt.internal.inc.IncrementalCompilerImpl.compileIncrementally(IncrementalCompilerImpl.scala:425)
[error] sbt.internal.inc.IncrementalCompilerImpl.compile(IncrementalCompilerImpl.scala:137)
[error] sbt.Defaults$.compileIncrementalTaskImpl(Defaults.scala:2371)
[error] sbt.Defaults$.$anonfun$compileIncrementalTask$2(Defaults.scala:2321)
[error] sbt.internal.server.BspCompileTask$.$anonfun$compute$1(BspCompileTask.scala:31)
[error] sbt.internal.io.Retry$.apply(Retry.scala:47)
[error] sbt.internal.io.Retry$.apply(Retry.scala:29)
[error] sbt.internal.io.Retry$.apply(Retry.scala:24)
[error] sbt.internal.server.BspCompileTask$.compute(BspCompileTask.scala:31)
[error] sbt.Defaults$.$anonfun$compileIncrementalTask$1(Defaults.scala:2319)
[error] scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:63)
[error] sbt.std.Transform$$anon$4.work(Transform.scala:69)
[error] sbt.Execute.$anonfun$submit$2(Execute.scala:283)
[error] sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:24)
[error] sbt.Execute.work(Execute.scala:292)
[error] sbt.Execute.$anonfun$submit$1(Execute.scala:283)
[error] sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error] sbt.CompletionService$$anon$2.call(CompletionService.scala:65)
[error] java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
[error] java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[error] java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[error] java.base/java.lang.Thread.run(Thread.java:833)
[error]            
[error] java.lang.AssertionError: assertion failed
[error] 	at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:11)
[error] 	at dotty.tools.dotc.core.TypeOps$.dominators$1(TypeOps.scala:248)
[error] 	at dotty.tools.dotc.core.TypeOps$.approximateOr$1(TypeOps.scala:382)
[error] 	at dotty.tools.dotc.core.TypeOps$.orDominator(TypeOps.scala:395)
[error] 	at dotty.tools.dotc.core.Types$OrType.join(Types.scala:3435)
[error] 	at dotty.tools.dotc.core.Types$Type.classSymbol(Types.scala:531)
[error] 	at dotty.tools.dotc.typer.Implicits$ImplicitSearch.<init>(Implicits.scala:1212)
[error] 	at dotty.tools.dotc.typer.Implicits.inferImplicit(Implicits.scala:1060)
[error] 	at dotty.tools.dotc.typer.Implicits.inferImplicit$(Implicits.scala:818)
[error] 	at dotty.tools.dotc.typer.Typer.inferImplicit(Typer.scala:116)
[error] 	at dotty.tools.dotc.typer.Implicits.inferImplicitArg(Implicits.scala:884)
[error] 	at dotty.tools.dotc.typer.Implicits.inferImplicitArg$(Implicits.scala:818)
[error] 	at dotty.tools.dotc.typer.Typer.inferImplicitArg(Typer.scala:116)
[error] 	at dotty.tools.dotc.typer.Typer.implicitArgs$1(Typer.scala:3737)
[error] 	at dotty.tools.dotc.typer.Typer.addImplicitArgs$1(Typer.scala:3773)
[error] 	at dotty.tools.dotc.typer.Typer.adaptNoArgsImplicitMethod$1(Typer.scala:3849)
[error] 	at dotty.tools.dotc.typer.Typer.adaptNoArgs$1(Typer.scala:4038)
[error] 	at dotty.tools.dotc.typer.Typer.adapt1(Typer.scala:4268)
[error] 	at dotty.tools.dotc.typer.Typer.adapt(Typer.scala:3587)
[error] 	at dotty.tools.dotc.typer.Typer.readapt$1(Typer.scala:3598)
[error] 	at dotty.tools.dotc.typer.Typer.adapt1(Typer.scala:4255)
[error] 	at dotty.tools.dotc.typer.Typer.adapt(Typer.scala:3587)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3184)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3188)
[error] 	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3300)
[error] 	at dotty.tools.dotc.typer.Typer.$anonfun$57(Typer.scala:2486)
[error] 	at dotty.tools.dotc.inlines.PrepareInlineable$.dropInlineIfError(PrepareInlineable.scala:243)
[error] 	at dotty.tools.dotc.typer.Typer.typedDefDef(Typer.scala:2486)
[error] 	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3024)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3111)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3184)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3188)
[error] 	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3210)
[error] 	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3256)
[error] 	at dotty.tools.dotc.typer.Typer.typedClassDef(Typer.scala:2669)
[error] 	at dotty.tools.dotc.typer.Typer.typedTypeOrClassDef$1(Typer.scala:3036)
[error] 	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3040)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3111)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3184)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3188)
[error] 	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3210)
[error] 	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3256)
[error] 	at dotty.tools.dotc.typer.Typer.typedPackageDef(Typer.scala:2812)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3081)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3112)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3184)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3188)
[error] 	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3300)
[error] 	at dotty.tools.dotc.typer.TyperPhase.typeCheck$$anonfun$1(TyperPhase.scala:44)
[error] 	at dotty.tools.dotc.typer.TyperPhase.typeCheck$$anonfun$adapted$1(TyperPhase.scala:54)
[error] 	at scala.Function0.apply$mcV$sp(Function0.scala:42)
[error] 	at dotty.tools.dotc.core.Phases$Phase.monitor(Phases.scala:440)
[error] 	at dotty.tools.dotc.typer.TyperPhase.typeCheck(TyperPhase.scala:54)
[error] 	at dotty.tools.dotc.typer.TyperPhase.runOn$$anonfun$3(TyperPhase.scala:88)
[error] 	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error] 	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error] 	at scala.collection.immutable.List.foreach(List.scala:333)
[error] 	at dotty.tools.dotc.typer.TyperPhase.runOn(TyperPhase.scala:88)
[error] 	at dotty.tools.dotc.Run.runPhases$1$$anonfun$1(Run.scala:246)
[error] 	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error] 	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error] 	at scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1321)
[error] 	at dotty.tools.dotc.Run.runPhases$1(Run.scala:262)
[error] 	at dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:270)
[error] 	at dotty.tools.dotc.Run.compileUnits$$anonfun$adapted$1(Run.scala:279)
[error] 	at dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:67)
[error] 	at dotty.tools.dotc.Run.compileUnits(Run.scala:279)
[error] 	at dotty.tools.dotc.Run.compileSources(Run.scala:194)
[error] 	at dotty.tools.dotc.Run.compile(Run.scala:179)
[error] 	at dotty.tools.dotc.Driver.doCompile(Driver.scala:37)
[error] 	at dotty.tools.xsbt.CompilerBridgeDriver.run(CompilerBridgeDriver.java:88)
[error] 	at dotty.tools.xsbt.CompilerBridge.run(CompilerBridge.java:22)
[error] 	at sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:91)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$7(MixedAnalyzingCompiler.scala:193)
[error] 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:248)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4(MixedAnalyzingCompiler.scala:183)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4$adapted(MixedAnalyzingCompiler.scala:163)
[error] 	at sbt.internal.inc.JarUtils$.withPreviousJar(JarUtils.scala:239)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:163)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:211)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:534)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:534)
[error] 	at sbt.internal.inc.Incremental$.$anonfun$apply$5(Incremental.scala:180)
[error] 	at sbt.internal.inc.Incremental$.$anonfun$apply$5$adapted(Incremental.scala:178)
[error] 	at sbt.internal.inc.Incremental$$anon$2.run(Incremental.scala:464)
[error] 	at sbt.internal.inc.IncrementalCommon$CycleState.next(IncrementalCommon.scala:116)
[error] 	at sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:56)
[error] 	at sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:52)
[error] 	at sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:263)
[error] 	at sbt.internal.inc.Incremental$.$anonfun$incrementalCompile$8(Incremental.scala:419)
[error] 	at sbt.internal.inc.Incremental$.withClassfileManager(Incremental.scala:506)
[error] 	at sbt.internal.inc.Incremental$.incrementalCompile(Incremental.scala:406)
[error] 	at sbt.internal.inc.Incremental$.apply(Incremental.scala:172)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:534)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:488)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCompilerImpl.scala:332)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.compileIncrementally(IncrementalCompilerImpl.scala:425)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.compile(IncrementalCompilerImpl.scala:137)
[error] 	at sbt.Defaults$.compileIncrementalTaskImpl(Defaults.scala:2371)
[error] 	at sbt.Defaults$.$anonfun$compileIncrementalTask$2(Defaults.scala:2321)
[error] 	at sbt.internal.server.BspCompileTask$.$anonfun$compute$1(BspCompileTask.scala:31)
[error] 	at sbt.internal.io.Retry$.apply(Retry.scala:47)
[error] 	at sbt.internal.io.Retry$.apply(Retry.scala:29)
[error] 	at sbt.internal.io.Retry$.apply(Retry.scala:24)
[error] 	at sbt.internal.server.BspCompileTask$.compute(BspCompileTask.scala:31)
[error] 	at sbt.Defaults$.$anonfun$compileIncrementalTask$1(Defaults.scala:2319)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:63)
[error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:69)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:283)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:24)
[error] 	at sbt.Execute.work(Execute.scala:292)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:283)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:65)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[error] 	at java.base/java.lang.Thread.run(Thread.java:833)
```
</details>

--- a/compiler/src/dotty/tools/dotc/core/Types.scala
+++ b/compiler/src/dotty/tools/dotc/core/Types.scala
@@ -657,6 +657,8 @@ object Types extends TypeUtils {
             tp.superType.baseClasses
           case tp: ClassInfo =>
             tp.cls.classDenot.baseClasses
+          case tp: WildcardType =>
+            tp.effectiveBounds.hi.baseClasses
           case _ => Nil
       catch case ex: Throwable =>
         handleRecursive("base classes of", this.show, ex)

--- a/tests/pos/i19460.scala
+++ b/tests/pos/i19460.scala
@@ -1,0 +1,5 @@
+type T[A] = A | Any
+
+def perform[A](using T[A]): A = perform2
+
+def perform2[A](using T[A]): A = ???

